File: VendingApp.Shared/Models/InventoryModel.cs
namespace VendingApp.Infrastructure.Models
{
    public class InventoryModel
    {
        public string ProductNr { get; set; }

        public string Name { get; set; }

        public int Quantity { get; set; }

        public decimal Price { get; set; }

        public decimal? TargetCurrencyRate { get; set; }

        public decimal PriceByRate => Price * TargetCurrencyRate.GetValueOrDefault(1);
    }
}

File: VendingApp.Web/ClientApp/src/app/models/inventory.model.ts

export class Inventory {
  constructor() {
    this.price = 0;
    this.priceByRate = 0;
  }

  public productNr: string;
  public name: string;
  public quantity: number;
  public price: number;
  public targetCurrencyRate?: number;
  public priceByRate: number;
}
